src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -118,6 +118,18 @@ internal static SafeFreeCredentials AcquireCredentialsHandle(string package, boo
                 ((inSecurityBufferArray != null && inSecurityBufferArray.Length != 0) ? inSecurityBufferArray[0] : null),
                 outSecurityBuffer,
                 ref contextFlags);
+
+            // Confidentiality flag should not be set if not requested
+            if(status == SecurityStatusPal.CompleteNeeded)
+            {
+                ContextFlagsPal mask = ContextFlagsPal.Confidentiality;
+                if((requestedContextFlags & mask) != (contextFlags & mask))
```

Comment:
Flags need not to be matched exactly, only the confidentiality flag need to be matched.
If there is mismatch in conf flag, we throw the protection level not supported error.